modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManager.java

```diff
@@ -358,6 +358,125 @@ public SCIMResponse listWithGET(UserManager userManager, String filter,
         }
     }
 
+    /**
+     * Method to list users at the Users endpoint.
+     * In the method, when the count is zero, the response will get zero results. When the count value is not
+     * specified (null) a default number of values for response will be returned. Any negative value to the count
+     * will return all the users.
+     *
+     * @param userManager       User manager
+     * @param filter            Filter to be executed
+     * @param startIndexInt     Starting index value of the filter
+     * @param countInt          Number of required results
+     * @param sortBy            SortBy
+     * @param sortOrder         Sorting order
+     * @param domainName        Domain name
+     * @param attributes        Attributes in the request
+     * @param excludeAttributes Exclude attributes
+     * @return SCIM response
+     */
+    @Override
+    public SCIMResponse listWithGET(UserManager userManager, String filter, Integer startIndexInt, Integer countInt,
+            String sortBy, String sortOrder, String domainName, String attributes, String excludeAttributes) {
+
+        FilterTreeManager filterTreeManager;
+        Node rootNode = null;
+        JSONEncoder encoder;
+        try {
+            int count = ResourceManagerUtil.processCount(countInt);
+            int startIndex = ResourceManagerUtil
+                    .processStartIndex(startIndexInt == null ? null : String.valueOf(startIndexInt));
+
+            if (sortOrder != null) {
```

Comment:
Shall we move these common codes to a new method and call it under both these listWithGET() methods?  else it will be a code duplication 
OR we can call `listWithGET(UserManager userManager, String filter, int startIndex, int count, String sortBy, String sortOrder, String domainName, String attributes, String excludeAttributes) `
under this new method

Reply:
Will be fixed in the next commit to the PR. It was decided to change the flow. Therefore a new fix is needed.